modules/services/getmail.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.getmail;
+
+  retrieverModule = types.submodule ({config,...}: {
+    options = {
+      type = mkOption {
+        type = types.enum [
+          "SimplePOP3Retriever"
+          "SimplePOP3SSLRetriever"
+          "SimpleIMAPRetriever"
+          "SimpleIMAPSSLRetriever"
+        ];
+        default = "SimpleIMAPSSLRetriever";
+        description = "Type of the retriever.";
+      };
+
+      server = mkOption {
+        type = types.string;
+        default = "";
+        description = "The remote server.";
+      };
+
+      username = mkOption {
+        type = types.string;
+        default = "";
+        description = "The server username.";
+      };
+
+      password = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = ''
+          The server password. Note that the passwords are stored clear in the
+          nix store, so it is recommended to not use this field, but instead
+          either leave empty or use <literal>passwordCommand</literal> instead.
+        '';
+      };
+
+      passwordCommand = mkOption {
+        type = types.nullOr (types.listOf types.string);
+        default = null;
+        example = ["${pkgs.gnupg}/bin/gpg" "--decrypt" "file.gpg"];
+        description = ''
+          The server password. With this the password is retrieved with the
+          given command. The list value is given escaped to the implementation.
+        '';
+      };
+
+      mailboxes = mkOption {
+        type = types.listOf types.string;
+        default = [];
+        description = "A list of mailboxes";
+      };
+    };
+  });
+
+  destinationModule = types.submodule ({config,...}: {
+    options = {
+      type = mkOption {
+        type = types.enum [
+          "MDA_external"
+          "Maildir"
+        ];
+        default = "Maildir";
+        description = "Destination type.";
+      };
+
+      path = mkOption {
+        type = types.string;
+        default = "$HOME/Mail";
+        example = "${pkgs.procmail}/bin/procmail";
+        description = ''
+          The destination path. For <literal>Maildir</literal> it's the file
+          path and for <literal>MDA_external</literal> it's the destination
+          application.
+        '';
+      };
+    };
+  });
+
+  optionsModule = types.submodule ({config,...}: {
```

Comment:
Since `config` isn't needed, this can be simplified to `types.submodule { options = ...; }`. The same for the other submodules.